File: backend/app/services/email_drafts.py
from app.models.operations import MatchDraft, Participant


def build_match_email(match: MatchDraft, participant: Participant, counterpart: Participant) -> tuple[str, str]:
    subject = "Your curated date information"
    counterpart_phone = (counterpart.phone or "").strip() or "phone not listed"
    body = (
        f"Hello {participant.first_name},\n\n"
        "The organizing team has prepared your date information for the upcoming session.\n\n"
        f"Date: {counterpart.first_name} {counterpart.last_name}\n"
        f"Phone: {counterpart_phone}\n\n"
        "Please follow the event instructions from the organizers and reach out to the team "
        "with any safety, scheduling, or discernment concerns.\n\n"
        "In Christ,\n"
        "The organizing team"
    )
    return subject, body

File: frontend/lib/emailDrafts.ts
export type DateEmailPartner = {
  name: string;
  phone?: string | null;
};

export function formatDraftPhone(phone?: string | null) {
  const trimmed = phone?.trim();
  return trimmed || "phone not listed";
}

export function buildParticipantDateEmail(firstName: string, partners: DateEmailPartner[]) {
  const dateList = partners
    .map((partner, i) => `${i + 1}. ${partner.name} - ${formatDraftPhone(partner.phone)}`)
    .join("\n");

  return {
    subject: "Your curated dates — Summer 2026",
    body: [
      `Hello ${firstName},`,
      "",
      "The organizing team has finalized your date assignments for the Summer 2026 session.",
      "",
      `Your date${partners.length > 1 ? "s" : ""}:`,
      dateList,
      "",
      "Please follow the event instructions from the organizers for each date and reach out with any questions before meeting.",
      "",
      "With care,",
      "The Little Dates Club Organizing Team",
    ].join("\n"),
  };
}
